src/lane.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "common.hpp"
+#include "car.hpp"
+
+class Lane : public Renderable
+{
+public:
+	// Creates all the associated render resources and default transform
+	bool init();
+
+	// Releases all associated resources
+	void destroy();
+
+	// Renders
+	void draw(const mat3& projection)override;
+
+	// Returns the current time remaining on the timer
+    float get_time_remaining() const;
+
+    // Returns the cars in the lane
+    Car* get_cars() const;
+
+private:
+	vec2 m_position; // Window coordinates
+	vec2 m_scale; // 1.f in each dimension. 1.f is as big as the associated texture
+	float m_rotation; // in radians
+	size_t m_num_indices; // passed to glDrawElements
+    Car* m_cars; // Cars in the lane

```

Comment:
Do we want to make this a vector instead of an array? As currently written, we have to define a size when we create this, whereas a vector functions more like a list in Python (will grow/shrink to accommodate values). An array might be nice if we want to have a strict limit to the number of cars that are allowed per lane, but if we do that, then we should have a variable (probably in this class) that exposes the max number of cars allowable per lane. 

Regardless of whether we use an array or a vector, we still need to initialize this variable in `init()` and get rid of it in `destroy()`.

For an array that looks like this:

`m_cars = new Car[5]` (or whatever size we want it) in `init()`
`delete[] m_cars` in `destroy()`

Syntax is slightly different for a vector, but I don't know it off the type of my head.

Reply:
I think we should have a max number of cars per lane. It'll help our AI because if one lane is already full, then the "hardest" option is to add to a different lane, and if all lanes are full, not adding more cars should at least give the player some chance to catch up. Besides, no point in wasting memory on a bunch of cars that are out of screen. I think a full lane should be one more car than the amount we want visible